app/wallets/BaseCosmosConnector.js

```diff
@@ -0,0 +1,265 @@
+
+import {
+  assetLists as chainAssets,
+} from 'chain-registry'
+
+export default class BaseCosmosConnector {
+  /**
+   * Constructor.
+   *
+   * @param {BaseCosmosConnectorParams} params - Parameters.
+   */
+  constructor ({
+    walletClerk,
+    chainWallet,
+    mainWallet,
+    chainName,
+    walletName,
+  }) {
+    this.walletClerk = walletClerk
+    this.chainWallet = chainWallet
+    this.mainWallet = mainWallet
+    this.chainName = chainName
+    this.walletName = walletName
+  }
+
+  /**
+   * Factory method to create a new instance of this class.
+   *
+   * @template {X extends typeof BaseCosmosConnector ? X : never} T, X
+   * @param {BaseCosmosConnectorFactoryParams} [params] - Parameters.
+   * @returns {InstanceType<T>} An instance of this class.
+   * @this {T}
+   */
+  static create ({
+    wallets = this.wallets,
+    chainName = this.chainName,
+    walletName = this.walletName,
+  } = {}) {
+    const walletClerk = this.createWalletClerk({
+      wallets,
+    })
+    const chainWallet = walletClerk.getChainWallet(
+      chainName,
+      walletName
+    )
+    const mainWallet = walletClerk.getMainWallet(walletName)
+
+    return /** @type {InstanceType<T>} */ (
+      new this({
+        walletClerk,
+        chainWallet,
+        mainWallet,
+        chainName,
+        walletName,
+      })
+    )
+  }
+
+  /**
+   * Create WalletClerk instance.
+   *
+   * @param {{
+   *   wallets: Array<import('@cosmos-kit/core').MainWalletBase>
+   * }} params - Parameters.
+   * @returns {InstanceType<typeof WalletClerk>}
+   */
+  static createWalletClerk ({
+    wallets,
+  }) {
+    const logger = this.createLogger()
+
+    return new WalletClerk(
+      this.supportedChains,
+      wallets,
+      logger,
+      false,
+      false,
+      [],
+      // @ts-expect-error: Upstream type mismatch.
+      chainAssets
+    )
+  }
+
+  /**
+   * Create logger for wallet clerk.
+   *
+   * @returns {InstanceType<typeof Logger>}
+   */
+  static createLogger () {
+    const logger = new Logger('NONE')
+
+    return logger
+  }
+
+  /**
+   * get: wallets
+   *
+   * @abstract
+   * @returns {Array<import('@cosmos-kit/core').MainWalletBase>} Array of MainWalletBase from cosmos-kit.
+   * @throws {Error} This function must be inherited.
+   */
+  static get wallets () {
+    throw new Error('this function must be inherited')
+  }
+
+  /**
+   * get: walletName
+   *
+   * @abstract
+   * @returns {string} Wallet name.
+   * @throws {Error} This function must be inherited.
+   */
+  static get walletName () {
+    throw new Error('this function must be inherited')
+  }
+
+  /**
+   * get: chainName
+   *
+   * @returns {string} Chain name.
+   */
+  static get chainName () {
+    return 'dydx'
+  }
+
+  /**
+   * get: supportedChains
+   *
+   * @returns {Array<import('@chain-registry/types').Chain | import('@cosmos-kit/core').ChainName>}
+   */
+  static get supportedChains () {
+    return [
+      'dydx',
+    ]
+  }
```

Comment:
@hecticme 

When both `'dydx'` string values have the same meaning, could define it as constant.

```
  static get chainName () {
    return 'dydx' // <---------------- (1)
  }

  /**  ... */
  static get supportedChains () {
    return [
      'dydx', // <-------------------- (2)
    ]
  }
```

```
/** @type {Record<string, import('@cosmos-kit/core').ChainName>>} */
const CHAIN_NAME = {
  DYDX: 'dydx',
}
```

Reply:
@hecticme 

If the name will be used in this class only, it is OK 1.

> export as a named export in this connector class.